<launch>

  <node pkg="sl_sensor_calibration" type="run_bundle_adjustment_node" name="run_bundle_adjustment_node" output="screen" required="true" >

    <param name="input_ba_problem_file" type="str" value="$(find sl_sensor_calibration)/bundle_adjustment_problem_files/ba_problem.txt" /> <!-- Where the initial BA problem file is located (generated by prepare_bundle_adjustment.launch) -->
    <param name="output_ba_problem_file" type="str" value="$(find sl_sensor_calibration)/bundle_adjustment_problem_files/ba_problem_final.txt" /> <!-- Where the initial BA problem file is located (generated by prepare_bundle_adjustment.launch) -->
    <param name="intrinsic_adjustment_mode" type="str" value="fixed" /> <!-- fixed, unconstrained or two_pass , refer to idiap/multicamera-calibration Github repo for more information -->
    <param name="residuals_save_folder" type="str" value="$(find sl_sensor_calibration)/residual_files/" />

    <!-- Where calibration files used for initialisation are located -->
    <param name="input_pri_cam_parameters_file" type="str" value="$(find sl_sensor_calibration)/calibration_files/primary_camera/primary_camera.xml" />
    <param name="input_sec_cam_parameters_file" type="str" value="$(find sl_sensor_calibration)/calibration_files/secondary_camera/secondary_camera.xml" />
    <param name="input_proj_parameters_file" type="str" value="$(find sl_sensor_calibration)/calibration_files/projector/projector.xml" />

    <!-- Where calibration files generated after BA will be saved -->
    <param name="output_pri_cam_parameters_file" type="str" value="$(find sl_sensor_calibration)/calibration_files/primary_camera/primary_camera_ba.xml" />
    <param name="output_sec_cam_parameters_file" type="str" value="$(find sl_sensor_calibration)/calibration_files/secondary_camera/secondary_camera_ba.xml" />
    <param name="output_proj_parameters_file" type="str" value="$(find sl_sensor_calibration)/calibration_files/projector/projector_ba.xml" />

  </node>

</launch>